app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -391,6 +391,11 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
                 view = view
             )
 
+            context.settings().setSitePermissionSettingListener(viewLifecycleOwner) {
+                this.context?.let { assignSitePermissionsRules(it) }
+            }
+            assignSitePermissionsRules(context)
```

Comment:
nit: I think we are calling `assignSitePermissionsRules` twice.

Reply:
We need to assign when first entering the fragment, and need to again whenever site permissions rules change (atm, this can happen when the user flips their wifi on/off from the browser fragment).  HOWEVER, it is calling twice every time the WIFI changes (because that updates AUTOPLAY_AUDIBLE and AUTOPLAY_INAUDIBLE).

I measured, and this assignment is taking ~.15 - .2MS on my Pixel 2.  Not a ton, but not nothing.  I spent some time trying to avoid the double call but haven't come up with a better solution.  Open to suggestions, but I don't think it's worth spending much more time on it.